src/lib/helpers/generateListenbrainzBody.ts

```diff
@@ -13,7 +13,10 @@ const generateListenbrainzBody = (body: Payload) => {
 					},
 					artist_name: body.Metadata.originalTitle ?? body.Metadata.grandparentTitle,
 					track_name: body.Metadata.title,
-					release_name: body.Metadata.parentTitle
+					release_name: body.Metadata.parentTitle,
+					...(body.Metadata?.Guid?.[0]?.id !== undefined && {
+                        track_mbid: body.Metadata.Guid[0].id.replace(/^mbid:\/\//, '')
```

Comment:
Just a few quick notes here!

1. The ListenBrainz payload allows a `recording_mbid` field, under `track_metadata` -> `additional_info`. See the [docs here](https://listenbrainz.readthedocs.io/en/latest/users/json.html#:~:text=A%20complete%20submit%20listen%20JSON%20document%20may%20look%20like%3A).

2. It doesn't look like this correctly handles situations where there's a GUID that isn't an `mbid`, or where the mbid isn't at the first array index. I know it seems like an unlikely edge case, but people using different agents for example may get a `plex://` guid, or even possibly a `lastfm://` one if I recall correctly. I'd suggest perhaps guarding against this with `Array.find`, looking for `id.startsWith('mbid')` for example. For readability, perhaps move this all up to the top of the function `const recording_mbid = ...`?

Reply:
> The ListenBrainz payload allows a recording_mbid field, under track_metadata -> additional_info. See the [docs here](https://listenbrainz.readthedocs.io/en/latest/users/json.html#:~:text=A%20complete%20submit%20listen%20JSON%20document%20may%20look%20like%3A).

Unfortunately we do not have the recording_mbid in the payload, only the track_mbid. Track MBIDs can be resolved into Recording+Release MBIDs via a call, but it's also supposedly supported in the LB payload.
I could add a lookup to the API, but that would be quite heavy, when in reality LB should just be supporting Track MBIDs out of the box.

> It doesn't look like this correctly handles situations where there's a GUID that isn't an mbid
Yup, edge case I didn't check correctly for. Thanks.

Reply:
Do you think regexing the UUID to make sure it's valid is also useful? I'll also use a match to make sure it's actually mbid://<valid uuid> before inserting it.

Reply:
Cleanup in https://github.com/simonxciv/eavesdrop.fm/pull/199/commits/aeec5f522f7d3fdc65359a4db83801b90cd05bda

Reply:
Fixed (again, for real) in f323df1b. This time I'm actually searching in the array.